modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheMetricsImpl.java

```diff
@@ -201,6 +208,9 @@
     /** Write-behind store, if configured. */
     private GridCacheWriteBehindStore store;
 
+    /** Tx collisions info. */
+    private volatile Supplier<List<Map.Entry<GridCacheMapEntry, Integer>>> txKeyCollisionInfo;
```

Comment:
I think it can be replaced with LinkedHashMap.

Reply:
don`t understand idea, each CacheMetricsImpl gets LinkedMap, what info would it map store ?

Reply:
I would prefer LinkedHashMap, it's up to you.

Reply:
What about adding comments like
`    private volatile Supplier<List<Map.Entry</* Collided keys. */ GridCacheMapEntry, /* Collisions queue size. */ Integer>>> txKeyCollisionInfo;` ?